src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -3225,24 +3225,26 @@ _mongoc_cluster_run_opmsg_send (
       message_length += mcd_rpc_header_set_response_to (rpc, 0);
       message_length += mcd_rpc_header_set_op_code (rpc, MONGOC_OP_CODE_MSG);
 
-      mcd_rpc_op_msg_set_sections_count (rpc, cmd->payload ? 2u : 1u);
+      BSON_ASSERT (cmd->payloads_count <= SIZE_MAX - 1u);
+      mcd_rpc_op_msg_set_sections_count (rpc, 1u + cmd->payloads_count);
 
       message_length += mcd_rpc_op_msg_set_flag_bits (rpc, flags);
       message_length += mcd_rpc_op_msg_section_set_kind (rpc, 0u, 0);
       message_length += mcd_rpc_op_msg_section_set_body (rpc, 0u, bson_get_data (cmd->command));
 
-      if (cmd->payload) {
-         BSON_ASSERT (bson_in_range_signed (size_t, cmd->payload_size));
+      for (size_t i = 0; i < cmd->payloads_count; i++) {
+         BSON_ASSERT (bson_in_range_signed (size_t, cmd->payloads[i].size));
```

Comment:
```suggestion
      for (size_t i = 0; i < cmd->payloads_count; i++) {
         const mongoc_cmd_payload_t* const payload = cmd->payloads[i];

         BSON_ASSERT (bson_in_range_signed (size_t, payload->size));
```
Recommend defining a `mongoc_cmd_payload_t` struct to allow for code such as suggested (avoiding repetitive subscripts).

Reply:
Fixed suggestion to take address of `.payloads` element. Alternatively:
```c
const mongoc_cmd_payload_t payload = cmd->payloads[i];

BSON_ASSERT (bson_in_range_signed (size_t, payload.size));
```
if the view-like ownership semantics of the local variable isn't considered too confusing.